torch/csrc/jit/mobile/type_parser.cpp

```diff
@@ -164,23 +167,190 @@ class TypeParser {
   size_t start_;
   std::string next_token_;
 };
+
+/*
+CustomTypeParser to parse custom type. Currently custom type will be
+in following structure:
+('type_name',
+    ('NamedTuple',
+        ('filed_name_1', 'field_type_1')
+        ('filed_name_2', 'field_type_2')
+    )
+)
+
+For example, python source code is:
+
+class Foo(NamedTuple):
+    id: torch.Tensor
+
+class Bar(torch.nn.Module):
+    def __init__(self):
+        super(Bar, self).__init__()
+        self.foo = Foo(torch.tensor(1))
+
+    def forward(self, a: torch.Tensor):
+        self.foo = Foo(a)
+        return self.foo
+
+The corresponding bytecode will be:
+(6,
+ ('__torch__.Bar.forward',
+  (('instructions',
+    (('STOREN', 1, 2),
+     ('LOAD', 1, 0),
+     ('MOVE', 2, 0),
+     ('NAMED_TUPLE_CONSTRUCT', 0, 1),
+     ('SET_ATTR', 2, 0),
+     ('MOVE', 1, 0),
+     ('GET_ATTR', 2, 0),
+     ('RET', 0, 0))),
+   ('operators', ()),
+   ('constants', ()),
+   ('types', (('__torch__.Foo', ('NamedTuple', (('id', 'Tensor'),))),)),
+   ('register_size', 2)),
+  (('arguments',
+    ((('name', 'self'), ('type', '__torch__.Bar'), ('default_value', None)),
+     (('name', 'a'), ('type', 'Tensor'), ('default_value', None)))),
+   ('returns',
+    ((('name', ''), ('type', 'Tuple[Tensor]'), ('default_value', None)),)))))
+
+*/
+class CustomTypeParser final {
```

Comment:
Have you run any BSB to check the size regress? Is it possible to put it in another file, so that it's easier for the modularization?

Reply:
Yeah for ios,
Compressed  : +3.8 KiB (+0.01%) => 73.69 MiB
Uncompressed: +8.7 KiB (+0.01%) => 161.83 MiB